calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java

```diff
@@ -142,6 +142,8 @@
 	
 	private Cache cache = null;
 	
+	private String m_calRefList = "";
```

Comment:
Using a Map to store a cache of values needs to be done very carefully as
it's really easy to leak memory or even if you don't leak, just use lots of
memory which is never recovered. For example in the current implementation
in this pull request if I make a request for a calendar that doesn't exist
the reference for that calendar will be store in the heap of the JVM
forever (until it's restarted), this make it a quick denial of service
attack.

On 31 May 2016 at 09:22, Daniel Merino Echeverría <notifications@github.com>
wrote:

> In
> calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java
> <https://github.com/sakaiproject/sakai/pull/2567#discussion_r65139971>:
>
> > @@ -142,6 +142,8 @@
> >  	
> >  	private Cache cache = null;
> >  	
> > +	private String m_calRefList = "";
>
> Yeah, thanks, Earle already explained me that. I have thinking how could I
> fix this and I think that the best choice would be to use a
> ConcurrentHashMap in BaseExternalCalendarSubscriptionService.java to store
> the references.
>
> If you know a better solution I am willing to know it.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sakaiproject/sakai/pull/2567/files/e4a2cb542155cdbe5eabbd2fe9f4c830b1b52e49#r65139971>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAXIWRc9wWkjdDhYcdcA8dC56d2Kg1kks5qG--3gaJpZM4IfN2h>
> .
>



-- 
  Matthew Buckett